compiler/packages/babel-plugin-react-compiler/src/Optimization/InlineJsxTransform.ts

```diff
@@ -0,0 +1,403 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  BuiltinTag,
+  Effect,
+  HIRFunction,
+  Instruction,
+  JsxAttribute,
+  makeInstructionId,
+  ObjectProperty,
+  Place,
+  SpreadPattern,
+} from '../HIR';
+import {
+  createTemporaryPlace,
+  markInstructionIds,
+  markPredecessors,
+  reversePostorderBlocks,
+} from '../HIR/HIRBuilder';
+
+function createSymbolProperty(
+  fn: HIRFunction,
+  instr: Instruction,
+  nextInstructions: Array<Instruction>,
+  propertyName: string,
+  symbolName: string,
+): ObjectProperty {
+  const symbolPlace = createTemporaryPlace(fn.env, instr.value.loc);
+  const symbolInstruction: Instruction = {
+    id: makeInstructionId(0),
+    lvalue: {...symbolPlace, effect: Effect.Mutate},
+    value: {
+      kind: 'LoadGlobal',
+      binding: {kind: 'Global', name: 'Symbol'},
+      loc: instr.value.loc,
+    },
+    loc: instr.loc,
+  };
+  nextInstructions.push(symbolInstruction);
+
+  const symbolForPlace = createTemporaryPlace(fn.env, instr.value.loc);
+  const symbolForInstruction: Instruction = {
+    id: makeInstructionId(0),
+    lvalue: {...symbolForPlace, effect: Effect.Read},
+    value: {
+      kind: 'PropertyLoad',
+      object: symbolInstruction.lvalue,
```

Comment:
note that places always have to be copied, not used by reference, so `{...symbolPlace}` here